From 03b6efd3153b9046f9e0e92403256d6109ee9a03 Mon Sep 17 00:00:00 2001
From: Jesse Chan <jc@lineageos.org>
Date: Sat, 21 Apr 2018 23:45:57 -0700
Subject: [PATCH 07/14] Add an option to force pre-O apps to use full screen
 aspect ratio

When an app target pre-O releases, the default max aspect ratio
is 1.86:1 which leads to ugly black areas on devices that have
screens with higher aspect ratio (for example Galaxy S8/S9).

This change adds an option to allow users to change default
aspect ratio for pre-O apps to 2.1 which would fit recent devices.

Change-Id: I88035fb079bad9a66b36fe5e861017af568b8f4c
---
 .../core/java/com/android/server/am/ActivityRecord.java   | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/am/ActivityRecord.java b/services/core/java/com/android/server/am/ActivityRecord.java
index 9723dc8442d..21ec9971a5d 100644
--- a/services/core/java/com/android/server/am/ActivityRecord.java
+++ b/services/core/java/com/android/server/am/ActivityRecord.java
@@ -131,6 +131,7 @@ import android.content.pm.ActivityInfo;
 import android.content.pm.ApplicationInfo;
 import android.content.res.CompatibilityInfo;
 import android.content.res.Configuration;
+import android.content.res.Resources;
 import android.graphics.Bitmap;
 import android.graphics.GraphicBuffer;
 import android.graphics.Rect;
@@ -351,6 +352,9 @@ final class ActivityRecord extends ConfigurationContainer implements AppWindowCo
     private boolean mShowWhenLocked;
     private boolean mTurnScreenOn;
 
+    private final float mFullScreenAspectRatio = Resources.getSystem().getFloat(
+                    org.lineageos.platform.internal.R.dimen.config_screenAspectRatio);
+
     /**
      * Temp configs used in {@link #ensureActivityConfigurationLocked(int, boolean)}
      */
@@ -2314,7 +2318,9 @@ final class ActivityRecord extends ConfigurationContainer implements AppWindowCo
     // TODO(b/36505427): Consider moving this method and similar ones to ConfigurationContainer.
     private void computeBounds(Rect outBounds) {
         outBounds.setEmpty();
-        final float maxAspectRatio = info.maxAspectRatio;
+        final float maxAspectRatio = (LineageSettings.System.getInt(service.mContext.
+                getContentResolver(), LineageSettings.System.FULL_SCREEN_ASPECT_RATIO, 0) != 0)
+                        ? mFullScreenAspectRatio : info.maxAspectRatio;
         final ActivityStack stack = getStack();
         if (task == null || stack == null || !task.mFullscreen || maxAspectRatio == 0
                 || isInVrUiMode(getConfiguration())) {
-- 
2.17.0

